File: invision-backend/app/Enums/PosTransactionType.php
<?php

namespace App\Enums;

enum PosTransactionType: string
{
    case SellOut = 'sell_out';
    case SellThrough = 'sell_through';
    case Return = 'return';

    public function label(): string
    {
        return match ($this) {
            self::SellOut => 'Sell Out',
            self::SellThrough => 'Sell Through',
            self::Return => 'Return',
        };
    }

    public function color(): string
    {
        return match ($this) {
            self::SellOut => 'blue',
            self::SellThrough => 'green',
            self::Return => 'red',
        };
    }
}

File: invision-mobile/lib/src/core/enums/pos_transaction_type.dart
enum PosTransactionType {
  sellOut,
  sellThrough,
  posReturn;

  static PosTransactionType fromString(String value) {
    return switch (value) {
      'sell_out' => PosTransactionType.sellOut,
      'sell_through' => PosTransactionType.sellThrough,
      'return' => PosTransactionType.posReturn,
      _ => PosTransactionType.sellOut,
    };
  }

  String get value => switch (this) {
    PosTransactionType.sellOut => 'sell_out',
    PosTransactionType.sellThrough => 'sell_through',
    PosTransactionType.posReturn => 'return',
  };

  String get label => switch (this) {
    PosTransactionType.sellOut => 'Sell Out',
    PosTransactionType.sellThrough => 'Sell Through',
    PosTransactionType.posReturn => 'Return',
  };
}
